Funcky/Extensions/EnumerableExtensions/MaxOrNone.cs

```diff
@@ -0,0 +1,335 @@
+        public static Option<TSource> MaxOrNone<TSource>(this IEnumerable<Option<TSource>> source, RequireClass<TSource>? ω = null)
+            where TSource : class
+            => source.WhereSelect(Identity).MaxOrNone(Identity);
+
+        /// <summary>
+        /// Returns the maximum value in a sequence of optional generic values compared by a Comparer{T}. If the sequence only consists of none or is empty it returns None.
+        /// </summary>
+        /// <param name="source">A sequence of optional generic values of TSource to determine the maximum value of.</param>
+        /// <param name="ω">Dummy Parameter, do not use.</param>
+        /// <returns>The maximum value in the sequence or None.</returns>
+        [Pure]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0027:Public API with optional parameter(s) should have the most parameters amongst its public overloads.", Justification = "The RequireStruct type is only for type deduction of the right overload.")]
+        public static Option<TSource> MaxOrNone<TSource>(this IEnumerable<Option<TSource>> source, RequireStruct<TSource>? ω = null)
+            where TSource : struct
+            => source.WhereSelect(Identity).MaxOrNone(Identity);
+
+        /// <summary>
+        ///  Invokes a transform function on each element of a sequence and returns the maximum from the generic values compared by a Comparer{T}. If the transforemd sequence only consists of none or is empty it returns None.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of source.</typeparam>
+        /// <param name="source">A sequence of generic values of TSource to determine the maximum value of.</param>
+        /// <typeparam name="TResult">The type of the value returned by selector.</typeparam>
+        /// <param name="selector">A transform function to apply to each element.</param>
+        /// <param name="ω">Dummy Parameter, do not use.</param>
+        /// <returns>The maximum value in the sequence or None.</returns>
+        [Pure]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0027:Public API with optional parameter(s) should have the most parameters amongst its public overloads.", Justification = "The RequireClass type is only for type deduction of the right overload.")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "The RequireClass type is needed on each overload.")]
+        public static Option<TResult> MaxOrNone<TSource, TResult>(this IEnumerable<TSource> source, Func<TSource, TResult> selector, RequireClass<TResult>? ω = null)
+            where TResult : class
+            => source.Aggregate(Option<TResult>.None(), (max, current) => Option.Some(max.Match(selector(current), m => Comparer<TResult>.Default.Compare(m, selector(current)) > 0 ? selector(current) : m)));
```

Comment:
`Comparer<TResult>.Default.Compare(m, selector(current)) > 0 ? selector(current) : m` could be extracted into a generic `Max` function

Reply:
Adressed in d194b57c